dbms/src/Storages/System/StorageSystemSettings.cpp

```diff
@@ -1,11 +1,25 @@
+#include <Storages/System/StorageSystemSettings.h>
 #include <DataTypes/DataTypeString.h>
+#include <DataTypes/DataTypeNullable.h>
 #include <DataTypes/DataTypesNumber.h>
 #include <Interpreters/Context.h>
-#include <Storages/System/StorageSystemSettings.h>
+#include <Access/SettingsConstraints.h>
 
 
 namespace DB
 {
+String StorageSystemSettings::valueToString(const StringRef & name, const Field & value)
+{
+    try
+    {
+        return Settings::getAsStringWithoutApplying(name, value);
```

Comment:
Why it can throw exception?

Reply:
You're right, it shouldn't throw exceptions. I've changed this code.